### PR TITLE
Feat/guppy api test plan

### PR DESCRIPTION
**Pull request by ZakirG**
This PR looks a little overwhelming, sorry. Start here: test_plans/guppy/test_plan.md 

This test plan is written in Gherkin (a gross terrible name for "structured executable English"); it tests the Guppy API from a user perspective. The "Background" sections each do the same thing, just with different test_data files. The Background section is to be executed before the Scenarios in a Feature. I technically violated Gherkin syntax in order to share setup commands between Features -- usually Background sections are within a Feature, I included mine before all Features and then referenced specific Backgrounds from within each Feature.

https://cucumber.io/docs/gherkin/reference/

I set out to test each feature mentioned in the Guppy documentation, with the exception of the accessibility parameter (which will be tested in the Tiered Access tests) and filterSelf. 

https://github.com/uc-cdis/guppy/blob/master/doc/queries.md

### CODE REVIEW

File: test_plans/guppy/test_plan.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		And match the contents of test_response_5.json
+
+Feature: A user can make requests exceeding 10k records to /download endpoint.
+	Background: (#2)
+
+	Scenario: Mingfei wants to execute the Scenario labeled (*) without worrying about paginating the results.
+		When Mingfei makes an API request to /guppy/download with test_query_8.json
+		Then the response will be successful with status code 200
+		And match the contents of test_response_8.json
+
```
Comment (atharvar28) [inline test_plans/guppy/test_plan.md]: @ZakirG I am not sure if you have merged test plan and gherkin feature file into one? I am not sure whether to review it as a test plan or as a gherkin file. The gherkin feature file has the extension '.feature' and we want to have both test plan as well as gherkin file for guppy (but separated).

File: test_plans/guppy/test_plan.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      (#2): |test_data/test_data_2.json|
+
+Feature: A user will be denied access to Guppy in the case that their auth is expired, and will be informed that their auth is invalid.
+	Background: (#1)
+
+	Scenario: Mingfei wants to make a query to Guppy, but his access token is expired.
+		Given mingfei@example.com's access token is expired
+		When Mingfei makes an API request to /guppy/graphql with test_query_1.json
+		Then the response will fail with a status code 401
+
```
Comment (atharvar28) [inline test_plans/guppy/test_plan.md]: @ZakirG you might want to use first-person pronouns in the scenarios.

File: test_plans/guppy/test_plan.md (hunk 1 above)
Comment (ZakirG) [reply test_plans/guppy/test_plan.md]: Oh I see ok. So I should make a separate test plan file in this format? https://github.com/uc-cdis/gen3-qa/blob/master/test_plans/pidgin/test_plan.md

File: test_plans/guppy/test_plan.md (hunk 2 above)
Comment (ZakirG) [reply test_plans/guppy/test_plan.md]: done!

**Pull request merged**